examples/debugger/indirect.c

```diff
@@ -389,17 +390,14 @@ fprintf(stderr, "DEBUGGER URI: %s\n", myuri);
     PMIX_INFO_LOAD(&info[n], PMIX_EVENT_HDLR_NAME, "LAUNCH-COMPLETE", PMIX_STRING);
     PMIx_Register_event_handler(&code, 1, info, 1,
                                 spawn_cbfunc, evhandler_reg_callbk, (void*)&mylock);
-    while (regpending && ilactive) {
-        struct timespec tp = {0, 500000};
-        nanosleep(&tp, NULL);
-    }
+    DEBUG_WAIT_THREAD(&mylock);
```

Comment:
You do have to watch out here. I think we used to have this statement early on, but you'll hang here if the IL aborts for some reason. I'd suggest going back to the `while` loop but correctly watching the `mylock.active` flag along with `ilactive`. It isn't as elegant as the macro with its `pthread_condition_wait`, but it prevents the hang.

Maybe somebody can come up with a way of doing the combined wait using the `condition_wait` method?

Reply:
Tell you what - let's take this as-is for now. We can defer this cleanup to the later stage as well.

Reply:
That sounds good. I'm working with @drwootton on cleaning up the examples for multi-node. I'll add that improvement to the list.